xcodeproj/internal/xcodeproj_rule.bzl

```diff
@@ -624,14 +638,31 @@ targets.
             for id in target_merge_srcs_by_label[src_target.label]:
                 target_merges.pop(id, None)
 
+    # Remap 'target_merge_dests' after popping invalid merges.
+    target_merge_dests = {}
     for src, dests in target_merges.items():
-        src_target = focused_targets.pop(src)
-
         for dest in dests:
-            focused_targets[dest] = xcode_targets.merge(
-                src = src_target,
-                dest = focused_targets[dest],
-            )
+            target_merge_dests.setdefault(dest, []).append(src)
+
+    for dest, srcs in target_merge_dests.items():
+        src_targets = [focused_targets.pop(src) for src in srcs]
+
+        # This functionality assumes that 2 or less sources are present in the
+        # potential merge. If that changes, this will need updated.
+        src_target_swift = None
+        src_target_non_swift = None
+
+        for src_target in src_targets:
+            if src_target._swift_params:
```

Comment:
I figure we can make `_swift_params` "public" in a follow up since it's now accessed here. Didn't want to complicate this already complex PR.